core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -772,7 +772,7 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
         .set(DRIVER_RESOURCES_FILE, resourcesFile)
         .setMaster("local-cluster[1, 1, 1024]")
         .setAppName("test-cluster")
-      conf.set(DRIVER_GPU_ID.amountConf, "1")
+      conf.set(DRIVER_GPU_ID.amountConf, "3")
```

Comment:
I think one of these was intentionally this way as it is allowed to find more resources then requested.  Perhaps put this back and add comment.

Reply:
The problem is, what does `sparkContext.resources()` should mean when we get Standalone resources scheduling in ? discovered resources or assigned resources ?  This changed to '3', otherwise it failed on:

`assert(sc.resources.get(GPU).get.addresses === Array("0", "1", "8"))` 

at L784.